File: src/main/java/um/haberes/core/model/view/TotalMensual.java
package um.haberes.core.model.view;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.Immutable;

import java.math.BigDecimal;

@Entity
@Table(name = "vw_total_mensual")
@Immutable
@Getter
@Setter
@EqualsAndHashCode
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class TotalMensual {

    @Id
    private String uniqueId = null;

    private Integer anho = null;

    private Integer mes = null;

    private Integer codigoId = null;

    private BigDecimal total = BigDecimal.ZERO;
}

File: src/main/java/um/haberes/core/kotlin/model/view/TotalMensual.kt
package um.haberes.core.kotlin.model.view

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.Immutable
import java.math.BigDecimal

@Entity
@Table(name = "vw_total_mensual")
@Immutable
data class TotalMensual(

    @Id
    var uniqueId: String? = null,

    var anho: Int? = null,
    var mes: Int? = null,
    var codigoId: Int? = null,
    var total: BigDecimal = BigDecimal.ZERO

)
